### ISSUE TITLE
Use window.matchMedia to automatically set Dark Mode

### ISSUE THREAD

**Issue by diballesteros** [diballesteros/nuzlocke#153]
**Is your feature request related to a problem? Please describe.**
Automatically detect dark mode

**Describe the solution you'd like**
Use window.matchMedia API to automatically set dark mode

**Comment by n-ce** [comment]
I have implemented this in my own [site](https://ytflix.netlify.app) however I don't have experience in Node or React, would I be assignable?

**Comment by diballesteros** [comment]
Yeah sure, go for it! If you need any help let me know. Should be a simple [useEffect](https://reactjs.org/docs/hooks-effect.html) hook. If you need any help let me know.

**Comment by n-ce** [comment]
Which file do I integrate the code in, (preferably the one that has the dark mode code)?

**Comment by n-ce** [comment]
Bro I have no idea how to make it however this is the code you require

let colorSchemeQueryList = window.matchMedia('(prefers-color-scheme: dark)');

const setColorScheme = e => {
  if (e.matches) {
//dark mode
  } else {
   //white mode
  }
}

setColorScheme(colorSchemeQueryList);
colorSchemeQueryList.addListener(setColorScheme);

**Comment by diballesteros** [comment]
> I have no idea how to implement it in your project however this is the code you require, Though I'm curious to see how it fits in your code.
> 
> ```
> let colorSchemeQueryList = window.matchMedia('(prefers-color-scheme: dark)');
> 
> const setColorScheme = e => {
>   if (e.matches) {
> //dark mode
>   } else {
>    //white mode
>   }
> }
> 
> setColorScheme(colorSchemeQueryList);
> colorSchemeQueryList.addListener(setColorScheme);
> ```

Looks about right! Without getting into too much detail React typically would handle this in a hook called `useEffect` on page load to change it at the beginning of the components [lifecycle](https://reactjs.org/docs/react-component.html).

There already is a boolean in the global state management (Zustand) that handles it called darkMode and a function that changes it called toggleMode

```ts
const useStore = create<AppState>(
  persist( // this change persists it to localstorage
    immer((set) => ({
      // ... OTHER STATE,
      darkMode: INITIAL_STATE.darkMode,
      toggleMode: () => {
        set((state) => {
          state.darkMode = !state.darkMode;
        });
      },
```
 
The primary calls to both of these parts of the state are located in the aforementioned `src/App.tsx`
 
 So possibly something like:
 
 ```ts
  useEffect(() => {
    if (window.matchMedia('(prefers-color-scheme: dark)') && !darkMode) {
      toggleMode();
    }
  }, []); // Empty to make sure it only runs once on pageload, if you're interested read up on useEffect to understand more
 ```
 Let me know if you have any more questions

**Comment by n-ce** [comment]
Yeah so am i supposed to only change the App.tsx file and put the code somehow in the useEffect section?

**Comment by diballesteros** [comment]
> Yeah so am i supposed to only change the App.tsx file and put the code somehow in the useEffect section?

Correct however its best practice to separate different concerns into different `useEffect`s. They execute in the order they're called. So this *new* `useEffect` should be above the one that already exists so that the user will have it automatically set before setting all the colors. 

Let me know how it feels when you test it out!
